addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -367,8 +367,8 @@ func (r *VSphereCPIConfigReconciler) getOwnerCluster(ctx context.Context, cpiCon
 	}
 	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: cpiConfig.Namespace, Name: clusterName}, cluster); err != nil {
 		if apierrors.IsNotFound(err) {
-			r.Log.Info(fmt.Sprintf("Cluster resource '%s/%s' not found", cpiConfig.Namespace, clusterName))
-			return nil, nil
+			r.Log.Error(err, fmt.Sprintf("Cluster resource '%s/%s' not found", cpiConfig.Namespace, clusterName))
```

Comment:
Should we ignore template objects? See https://github.com/vmware-tanzu/tanzu-framework/pull/2721/files

Reply:
@HanFa  can you follow up on this please?